lib/AST/Macro.cpp

```diff
@@ -880,6 +880,108 @@ MacroRange MacroSubstitution::ReplacementChildren(void) const noexcept {
   }
 }
 
+// Returns the first and last fully substituted tokens from a given macro
+// substitution, if they exist.
+std::pair<std::optional<Token>, std::optional<Token>>
+GetFirstAndLastFullySubstitutedTokens(
+  MacroSubstitution const &substitution) noexcept {
+
+  // Get the fully substituted first token
+  std::optional<Token> first_tok = std::nullopt;
+  {
+    auto subst = std::optional(substitution);
+    auto first_replacement_child = std::optional<Macro>();
+    while (subst && !subst->ReplacementChildren().empty()) {
```

Comment:
Save return value of `ReplacementChildren()` to a local var so you can reuse it.